### DISCUSSION TITLE
Workerd specification inquiry

### DISCUSSION THREAD

**Discussion by NoahHahm** [cloudflare/workerd#1324] [Q&A] [answered]
I understood Workerd's project as a server-based project.
Therefore, the entry point to every system must implement the fetch function.
Is there a protocol that can run scripts like nodejs' worker_thread function and send/receive them like IPC messages?
I wonder if I should send/receive using a web socket using durable-objects.

I would like to ask if there is anything that can be of help.

**Comment by jasnell** [comment, marked as answer]
> Is there a protocol that can run scripts like nodejs' worker_thread function and send/receive them like IPC messages?

The quick answer is no, there is no support for anything like Node.js' worker_threads and IPC. The longer answer is that you could use `connect()` to establish a TCP connection with a Node.js process and handle much of the communication yourself but that's about the extent. The options immediately available built-in to `workerd` are http(s) and websockets.
